ocaml-lsp-server/src/custom_requests/req_polarity_search.ml

```diff
@@ -0,0 +1,124 @@
+open Import
+module TextDocumentPositionParams = Lsp.Types.TextDocumentPositionParams
+
+let meth = "ocamllsp/polaritySearch"
+let capability = "handlePolaritySearch", `Bool true
+
+let completion_kind kind : CompletionItemKind.t =
+  match kind with
+  | `Value -> Value
+  | `Variant -> EnumMember
+  | `Label -> Field
+  | `Module -> Module
+  | `Modtype -> Interface
+  | `MethodCall -> Method
+  | `Keyword -> Keyword
+  | `Constructor -> Constructor
+  | `Type -> TypeParameter
+;;
+
+let kind_of_completion_kind (kind : CompletionItemKind.t) =
+  match kind with
+  | Value -> `Value
+  | EnumMember -> `Variant
+  | Field -> `Label
+  | Module -> `Module
+  | Interface -> `Modtype
+  | Method -> `MethodCall
+  | Keyword -> `Keyword
+  | Constructor -> `Constructor
+  | TypeParameter -> `Type
+  | _ -> failwith "Unknown kind"
+;;
+
+module PolaritySearchParams = struct
+  type t =
+    { text_document : TextDocumentIdentifier.t
+    ; position : Position.t
+    ; query : string
+    }
+
+  let t_of_yojson json =
+    let open Yojson.Safe.Util in
+    let textDocumentPosition = Lsp.Types.TextDocumentPositionParams.t_of_yojson json in
+    let query = json |> member "query" |> to_string in
+    { position = textDocumentPosition.position
+    ; text_document = textDocumentPosition.textDocument
+    ; query
+    }
+  ;;
+
+  let yojson_of_t { text_document; position; query } =
+    `Assoc
+      (("textDocument", TextDocumentIdentifier.yojson_of_t text_document)
+       :: ("position", Position.yojson_of_t position)
+       :: [ "query", `String query ])
+  ;;
+end
+
+module PolaritySearch = struct
+  type t = Query_protocol.Compl.entry list
+
+  let entry_of_yojson json =
+    let open Yojson.Safe.Util in
+    let name = json |> member "name" |> to_string in
+    let kind =
+      json |> member "kind" |> CompletionItemKind.t_of_yojson |> kind_of_completion_kind
+    in
+    let desc = json |> member "desc" |> to_string in
+    let info = json |> member "info" |> to_string in
+    let deprecated = json |> member "deprecated" |> to_bool in
+    { Query_protocol.Compl.name; kind; desc; info; deprecated }
+  ;;
```

Comment:
Do we really need to have the `*_of_yojson` versions of these object we never need to decode ? Is it to fullfil an imposed interface ? (same question for the following t_of_yojson).

Reply:
You don't need to write encoder/decoders if you don't use them. `ppx_yojson_conv` allows you to generate converters one way. It's only necessary to make sure the conventions of `ppx_yojson_conv` are respected so all decoders/encoders compose whether they are generated or not.